core/modules/kassyModule.js

```diff
@@ -0,0 +1,149 @@
+    files           = require.once('./../files.js'),
+    config          = require('./../config.js'),
+    modulesDir      = 'modules',
+    descriptor      = 'kassy.json';
+
+var verifyModuleDescriptior = function (kj, disabled) {
+    if (!kj.name || !kj.startup || !kj.version) {
+        return false;
+    }
+
+    if (disabled === true && exports.disabledConfig &&
+        exports.disabledConfig[kj.name] && exports.disabledConfig[kj.name] === true) {
+        return false;
+    }
+    return true;
+};
+
+exports.verifyModule = function (location, disabled) {
+    var stat = fs.statSync(location);
+    if (!stat.isDirectory()) {
+        return null;
+    }
+
+    var folderPath = path.resolve(location),
+        p = path.join(folderPath, './' + descriptor);
+    try {
+        stat = fs.statSync(p);
+        if (!stat) {
+            return null;
+        }
+    }
+    catch (e) {
+        return null;
+    }
+
+    var kj = require.once(p);
+    if (!verifyModuleDescriptior(kj, disabled)) {
+        return null;
+    }
+
+    if (!kj.folderPath) {
+        kj.folderPath = folderPath;
+    }
+    return kj;
+};
+
+exports.listModules = function (disabled) {
+    var data = files.filesInDirectory('./' + modulesDir),
+        modules = {};
+
+    for (var i = 0; i < data.length; i++) {
+        try {
+            var candidate = path.resolve(path.join(modulesDir, data[i])),
+                output = exports.verifyModule(candidate, disabled);
+            if (output) {
+                modules[output.name] = output;
+            }
+            else {
+                console.debug('Skipping "' + data[i] + '". It isn\'t a Kassy module.');
+            }
+        } catch (e) {
+            console.debug('A failure occured while listing "' + data[i] + '". It doesn\'t appear to be a module.');
+            console.critical(e);
+            continue;
+        }
+    }
+    return modules;
+};
+
+var createHelp = function(module) {
+    return function(commandPrefix) {
+        var h = [];
+        for (var i = 0; i < module.help.length; i++) {
+            var l = [];
+            for (var j = 0; j < module.help[i].length; j++) {
+                l.push(module.help[i][j].replace(/{{commandPrefix}}/g, commandPrefix));
+            }
+            h.push(l);
+        }
+        return h;
+    };
+},
+
+createMatcher = function(module) {
+    return function(event, commandPrefix) {
+        return event.arguments[0] === commandPrefix + module.command;
+    };
+},
+
+createHistoricalMatcher = function(matcher) {
+    return function(event, commandPrefix) {
+        return matcher(event.body, commandPrefix);
+    };
+},
+
+getFunctionParameterNames = function (func) {
+    // http://stackoverflow.com/questions/9091838/get-function-parameter-names-for-interface-purposes
+    var f = func.toString();
+    return f.match(/\(.*?\)/)[0].replace(/[()]/gi, '').replace(/\s/gi, '').split(',');
+};
+
+exports.loadModule = function (module) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962470/issues/source?bid=3282987&fileBranchId=3282998#l115)